src/test/steps/loginSteps.ts

```diff
@@ -0,0 +1,35 @@
+import { Given, When, Then } from "@cucumber/cucumber";
+import { chromium, Page, Browser } from "@playwright/test";
+
+let browser: Browser;
+let page: Page;
+
+Given('User navigates to the application', async function () {
+    browser = await chromium.launch({headless: false});
+    page = await browser.newPage();
+    await page.goto("https://px-stage.self-service.danads.com");
+});
+
+Given('User clicks login button', async function () {
+    //await page.locator("//a/*[text()='Log in']").click();
+})
+
+Given('User enters username', async function () {
+    page.locator("(//input[@name='username'])[2]").type("Username");
```

Comment:
Locators should be stored in page objects, aren't they?

Reply:
Yes, they should be. But this is just a sample test script.